internal/client/p.go

```diff
@@ -175,6 +175,46 @@ func (pc *p) CreateSubnet(
 	return txID, took, err
 }
 
+func (pc *p) getValidator(nodeID ids.ShortID) (start time.Time, end time.Time, found bool, err error) {
+	// make sure NodeID is a staker on the primary network
+	// TODO: official wallet client should define the error value for such case
+	// currently just returns "staking too short"
+	vs, err := pc.Client().GetCurrentValidators(constants.PrimaryNetworkID, []ids.ShortID{})
+	if err != nil {
+		return time.Time{}, time.Time{}, found, err
+	}
+	if len(vs) < 1 {
+		return time.Time{}, time.Time{}, found, ErrEmptyValidator
+	}
+	var validator map[string]interface{}
+	for _, v := range vs {
+		va, ok := v.(map[string]interface{})
+		if !ok {
+			return time.Time{}, time.Time{}, found, ErrNotValidatingPrimaryNetwork
+		}
+		nodeIDInf := va["nodeID"]
+		nodeIDs, _ := nodeIDInf.(string)
+		if nodeIDs == nodeID.PrefixedString(constants.NodeIDPrefix) {
+			validator = va
+			found = true
+			break
+		}
+	}
+	if !found {
+		return time.Time{}, time.Time{}, found, ErrNotValidatingPrimaryNetwork
+	}
+
+	dur := validator["startTime"]
```

Comment:
nit: `dur, ok := validator["startTime"].(int64)`

will return !ok if not in map or not `int64`